Source/CombatExtended/CombatExtended/CE_Utility.cs

```diff
@@ -633,6 +633,8 @@ public static void ThrowEmptyCasing(Vector3 loc, Map map, FleckDef casingFleckDe
             {
                 return;
             }
+
+            if (MP.IsInMultiplayer) Rand.PushState();
```

Comment:
The performance wasn't really the thing I had in mind when doing this. Doing a `PushState` puts the current RNG state into a stack, and when you pop it then it'll be restored to what it was before. I've been told this may end up making the RNG feel less random. However, it may not be a big issue here.

So if you leave me a response to confirm, I'll remove all the branching there and only leave push/pop state.

Reply:
I was going to inquire about how PushState behaves with regard to randomness.  If I understand correctly, the flek maker does *not* need to be synchronized between different players, hence why the Rand's state gets pushed and then restored.  Why not initialzie a new system random source that does not affect `MP.Rand`'s state and use that for non-synchronized things?

Reply:
The issue is what's going on inside of RimWorld code. `FleckMaker` class is doing some randomness inside of it, which is causing issues with Multiplayer, and considering it's not inside code we control - we can't really replace RNG calls in there.